src/main/java/com/playmonumenta/networkrelay/RemotePlayerManager.java

```diff
@@ -0,0 +1,347 @@
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class RemotePlayerManager implements Listener {
+	public static final String REMOTE_PLAYER_CHANNEL_BASE = "monumenta.redissync.remote_player";
+	public static final String REMOTE_PLAYER_UPDATE_CHANNEL = REMOTE_PLAYER_CHANNEL_BASE + ".update";
+	public static final String REMOTE_PLAYER_REFRESH_CHANNEL = REMOTE_PLAYER_CHANNEL_BASE + ".refresh";
+
+	@MonotonicNonNull
+	private static RemotePlayerManager INSTANCE = null;
+
+	private static final Map<String, Map<String, RemotePlayer>> mRemotePlayerShardMapped = new ConcurrentSkipListMap<>();
+	private static final Map<UUID, RemotePlayer> mRemotePlayersByUuid = new ConcurrentSkipListMap<>();
+	private static final Map<String, RemotePlayer> mRemotePlayersByName = new ConcurrentSkipListMap<>();
+
+
+	private final NetworkRelay mPlugin;
+	protected RemotePlayerManager(NetworkRelay plugin) {
+        mPlugin = plugin;
+		String lShard = getShardName();
+		try {
+			for (String shard: NetworkRelayAPI.getOnlineShardNames()) {
+				if (shard.equals(lShard)) {
+					continue;
+				}
+				mRemotePlayerShardMapped.put(shard, new ConcurrentSkipListMap<>());
+			}
+		} catch (Exception ex) {
+			mPlugin.getLogger().warning("Failed to get load shards");
+		}
+
+		try {
+			NetworkRelayAPI.sendBroadcastMessage(REMOTE_PLAYER_REFRESH_CHANNEL, new JsonObject());
+		} catch (Exception e) {
+			mPlugin.getLogger().warning("Failed to broadcast to channel " + REMOTE_PLAYER_REFRESH_CHANNEL);
+        }
+
+		INSTANCE = this;
+    }
+
+	@Nullable
+	protected static RemotePlayerManager getInstance() {
+		if (INSTANCE == null) {
+			return null;
+		}
+		return INSTANCE;
+	}
+
+	protected String[] getAllOnlinePlayersName() {
+		return mRemotePlayersByName.keySet().toArray(new String[0]);
+	}
+
+	@Nullable
+	protected String getPlayerShard(@NotNull String username) {
+		@Nullable RemotePlayer remotePlayer = getRemotePlayer(username);
+		if (remotePlayer == null) {
+			return null;
+		}
+		return remotePlayer.mShard;
+	}
+
+	@Nullable
+	protected String getPlayerShard(@NotNull UUID playerUuid) {
+		@Nullable RemotePlayer remotePlayer = getRemotePlayer(playerUuid);
+		if (remotePlayer == null) {
+			return null;
+		}
+		return remotePlayer.mShard;
+	}
+
+	@Nullable
+	protected RemotePlayer getRemotePlayer(@NotNull String username) {
+		return getRemotePlayer(getPlayerUuid(username));
+	}
+
+	@Nullable
+	protected RemotePlayer getRemotePlayer(@Nullable UUID playerUuid) {
+		if (playerUuid == null) {
+			return null;
+		}
+
+		@Nullable RemotePlayer remotePlayer = mRemotePlayersByUuid.get(playerUuid);
+		return remotePlayer;
+	}
```

Comment:
This whole block is weird - seems like playerUuid should be not nullable (i.e. no annotation), and just return directly `mRemovePlayersByUuid.get(playerUuid)` without the temporary variable.